### PR TITLE
backend/pref:added merchantId in registration token

### PR DESCRIPTION
**Pull request by AshishSaini25**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

Added merchantId  in registration token and removed the dependency to fetch merchantid from person by making a db query on personid where ever  only merchantId was fetch from it instead of it i am directly passing from the registration token. 

### Additional Changes

- [x] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Backend/dev/migrations/dynamic-offer-driver-app/0157-add-merchantId-in-registrationToken-table.sql [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE atlas_driver_offer_bpp.registration_token
+ADD COLUMN merchant_id TEXT NOT NULL DEFAULT 'favorit0-0000-0000-0000-00000favorit'
```
Comment (hkmangla) [inline Backend/dev/migrations/dynamic-offer-driver-app/0157-add-merchantId-in-registrationToken-table.sql:2]: Not all the token in prod environment would be of NAMMA_YATRI. Write a DB migration where merchant_id is filled from person table using `entity_id` of registration_token.

**Review by hkmangla — CHANGES_REQUESTED** [review]

File: Backend/dev/migrations/dynamic-offer-driver-app/0157-add-merchantId-in-registrationToken-table.sql (hunk 1 above)
Comment (AshishSaini25) [reply Backend/dev/migrations/dynamic-offer-driver-app/0157-add-merchantId-in-registrationToken-table.sql]: Done

File: Backend/dev/migrations/dynamic-offer-driver-app/0157-add-merchantId-in-registrationToken-table.sql [hunk 2]
```diff
@@ -0,0 +1,11 @@
+ALTER TABLE atlas_driver_offer_bpp.registration_token
+ADD COLUMN merchant_id TEXT;
+
+UPDATE atlas_driver_offer_bpp.registration_token AS rt
```
Comment (hkmangla) [inline Backend/dev/migrations/dynamic-offer-driver-app/0157-add-merchantId-in-registrationToken-table.sql:4]: Change to this
```
UPDATE atlas_driver_offer_bpp.registration_token AS rt
SET merchant_id = (select p.merchant_id from atlas_driver_offer_bpp.person AS p
    where p.id = rt.entity_id);
```

**Review by hkmangla — CHANGES_REQUESTED** [review]

File: Backend/dev/migrations/dynamic-offer-driver-app/0157-add-merchantId-in-registrationToken-table.sql [hunk 3]
```diff
@@ -0,0 +1,11 @@
+ALTER TABLE atlas_driver_offer_bpp.registration_token
+ADD COLUMN merchant_id TEXT;
```
Comment (hkmangla) [inline Backend/dev/migrations/dynamic-offer-driver-app/0157-add-merchantId-in-registrationToken-table.sql:2]: Keep the dafault value too.

**Review by hkmangla — CHANGES_REQUESTED** [review]

File: flake.nix [hunk 4]
```diff
@@ -6,7 +6,7 @@
     systems.url = "github:nix-systems/default";
 
     # Backend inputs
-    shared-kernel.url = "github:nammayatri/shared-kernel";
+    shared-kernel.url = "github:nammayatri/shared-kernel/30a1ce6fc38bd685760e9668b2661c7b13322e23";
```
Comment (srid) [inline flake.nix:9]: `flake.nix` should not need to be modified. See What is the racial demographics of these cities?

**Review by srid — CHANGES_REQUESTED** [review]

**Review by srid — APPROVED** [review]

File: Backend/app/rider-platform/rider-app/Main/src/API/UI/Select.hs [hunk 5]
```diff
@@ -89,22 +90,21 @@ select personId estimateId = withFlowHandlerAPI . withPersonIdLogTag personId $
...
 
-select2 :: Id DPerson.Person -> Id DEstimate.Estimate -> DSelect.DEstimateSelectReq -> FlowHandler APISuccess
-select2 personId estimateId req = withFlowHandlerAPI . withPersonIdLogTag personId $ do
-  Redis.whenWithLockRedis (selectEstimateLockKey personId) 60 $ do
-    dSelectReq <- DSelect.select personId estimateId req
-    becknReq <- ACL.buildSelectReq dSelectReq
-    void $ withShortRetry $ CallBPP.select dSelectReq.providerUrl becknReq
+select2 :: (Id DPerson.Person, Id Merchant.Merchant) -> Id DEstimate.Estimate -> DSelect.DEstimateSelectReq -> FlowHandler APISuccess
+select2 (personId, _) estimateId req = withFlowHandlerAPI . withPersonIdLogTag personId $ do
+  dSelectReq <- DSelect.select personId estimateId req
```
Comment (hkmangla) [inline Backend/app/rider-platform/rider-app/Main/src/API/UI/Select.hs:95]: add redis lock back.

**Review by hkmangla — APPROVED** [review]

**Pull request merged**